EpikCord/client/commands.py

```diff
@@ -49,17 +52,455 @@ def to_dict(self):
             payload[
                 "default_member_permissions"
             ] = self.default_member_permissions.value
-        if self.guild_only is not None:
-            payload["dm_permission"] = not self.guild_only
         if name_localizations := localization_list_to_dict(
             self.name_localizations
         ):
             payload["name_localizations"] = name_localizations
-        # if description_localizations := localization_list_to_dict(self.description_localizations):
-        #     payload["description_localizations"] = description_localizations
 
         return payload
 
 
 class ClientContextMenuCommand(BaseClientCommand):
     ...
+
+class ClientUserCommand(ClientContextMenuCommand):
+    def __init__(
+        self,
+        name: str,
+        callback: AsyncFunction,
+        *,
+        guild_ids: List[int] = [],
+        name_localizations: List[Localization] = [],
+        guild_only: Optional[bool] = None,
+        default_member_permissions: Optional[Permissions] = None,
+        nsfw: bool = False,
+    ):
+        super().__init__(
+            name,
+            callback,
+            guild_ids=guild_ids,
+            name_localizations=name_localizations,
+            type=ApplicationCommandType.USER,
+            guild_only=guild_only,
+            default_member_permissions=default_member_permissions,
+            nsfw=nsfw,
+        )
+
+class ClientMessageCommand(ClientContextMenuCommand):
+    def __init__(
+        self,
+        name: str,
+        callback: AsyncFunction,
+        *,
+        guild_ids: List[int] = [],
+        name_localizations: List[Localization] = [],
+        guild_only: Optional[bool] = None,
+        default_member_permissions: Optional[Permissions] = None,
+        nsfw: bool = False,
+    ):
+        super().__init__(
+            name,
+            callback,
+            guild_ids=guild_ids,
+            name_localizations=name_localizations,
+            type=ApplicationCommandType.MESSAGE,
+            guild_only=guild_only,
+            default_member_permissions=default_member_permissions,
+            nsfw=nsfw,
+        )
+
+class ApplicationCommandOptionChoice:
+    def __init__(self, name: str, value: str, *, name_localizations: List[Localization] = []):
+        self.name = name
+        self.value = value
+        self.name_localizations = name_localizations
+
+    def to_dict(self):
+        payload: Dict[str, Union[str, Dict[str, str]]] = {
+            "name": self.name,
+            "value": self.value,
+        }
+
+        if name_localizations := localization_list_to_dict(
+            self.name_localizations
+        ):
+            payload["name_localizations"] = name_localizations
+
+        return payload
+
+
+class BaseApplicationCommandOption:
```

Comment:
All classes with a `to_dict` should inherit from some kind of `APIConvertible` or similar ABC or Protocol

Reply:
For what benefit?

Reply:
It force you to implement the abstract member, creating a interface for that have meaning, where you can expect class based on it to have the required methods.